### PR TITLE
Miscellaneous cleanup/improvements

### PR DESCRIPTION
**Pull request by jessebraham**
- Add descriptions for command-line arguments
- Use `?` in favour of `unwrap`
- Small cleanup/refactoring

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -243,57 +252,53 @@ fn main() {
...
-    let dir = path.join(&args.name);
-    std::fs::create_dir(&dir).unwrap();
+    let project_dir = path.join(&args.name);
+    fs::create_dir(&project_dir)?;
 
     for &(file_path, contents) in template_files::TEMPLATE_FILES.iter() {
-        let path = dir.join(file_path);
+        if file_path == "Cargo.lock" {
+            continue;
+        }
```
Comment (jessebraham) [inline src/main.rs:261]: Is this check actually required? I don't see `Cargo.lock` listed in `TEMPLATE_FILES`

File: src/main.rs [hunk 2]
```diff
@@ -163,22 +166,28 @@ static CHIP_VARS: &[(Chip, &[(&str, &str)])] = &[
...
 
+    /// Chip to target
     #[arg(short, long)]
     chip: Chip,
 
+    /// Run in headless mode (i.e. do not use the TUI)
     #[arg(long)]
     headless: bool,
 
+    // TODO: Can we list the options and/or point users to some documentation?
```
Comment (SergioGasquez) [inline src/main.rs:180]: This is probably worth an issue, readme has the list of options but not sure if that is the best place...

File: src/main.rs (hunk 1 above)
Comment (SergioGasquez) [reply src/main.rs:261]: I think this can be removed, the template/Cargo.lock is git ignored.

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks

File: src/main.rs (hunk 2 above)
Comment (jessebraham) [reply src/main.rs:180]: Created #32

**Pull request merged**